api/v1alpha1/capp_types.go

```diff
@@ -24,38 +24,93 @@ import (
 
-// CappSpec defines the desired state of Capp
+// CappSpec defines the desired state of Capp.
 type CappSpec struct {
-	ScaleMetric       string                      `json:"scaleMetric,omitempty"`
-	Site              string                      `json:"site,omitempty"`
+	// ScaleMetric defines which metric type is watched by the Autoscaler.
+	// Possible values examples: "concurrency", "rps", "cpu", "memory".
+	// +optional
+	ScaleMetric string `json:"scaleMetric,omitempty"`
+
+	// Site defines where to deploy the Capp.
+	// It can be a specific cluster or a placement name.
+	// +optional
+	Site string `json:"site,omitempty"`
+
+	// ConfigurationSpec holds the desired state of the Configuration (from the client).
 	ConfigurationSpec knativev1.ConfigurationSpec `json:"configurationSpec"`
-	RouteSpec         RouteSpec                   `json:"routeSpec,omitempty"`
+
+	// RouteSpec defines the route specification for the Capp.
+	// +optional
+	RouteSpec RouteSpec `json:"routeSpec,omitempty"`
 }
 
+// RouteSpec defines the route specification for the Capp.
 type RouteSpec struct {
+	// Hostname is a custom DNS name for the Capp route.
+	// +optional
 	Hostname string `json:"hostname,omitempty"`
-	TlsEnabled    bool   `json:"tlsEnabled,omitempty"`
+
+	// TlsEnabled determines whether to enable TLS for the Capp route.
+	// +optional
+	TlsEnabled bool `json:"tlsEnabled,omitempty"`
+
+	// TrafficTarget holds a single entry of the routing table for the Capp route.
+	// +optional
+	TrafficTarget knativev1.TrafficTarget `json:"trafficTarget,omitempty"`
+
+	// RouteTimeoutSeconds is the maximum duration in seconds
+	// that the request instance is allowed to respond to a request.
+	// +optional
 	RouteTimeoutSeconds *int64 `json:"routeTimeoutSeconds,omitempty"`
-	//TrafficTarget knativev1.TrafficTarget `json:"trafficTarget,omitempty"`
 }
 
+// ApplicationLinks contains relevant information about
+// the cluster that the Capp was deployed in.
 type ApplicationLinks struct {
-	ConsoleLink    string `json:"consoleLink,omitempty"`
-	Site           string `json:"site,omitempty"`
+	// ConsoleLink holds a link for the cluster console.
+	// +optional
+	ConsoleLink string `json:"consoleLink,omitempty"`
+
+	// Site holds the cluster name that the Capp was deployed on.
+	// +optional
+	Site string `json:"site,omitempty"`
+
+	// ClusterSegment holds the segment of the cluster
+	// that the Capp was deployed on.
+	// Can be used for opening ports, etc.
+	// +optional
 	ClusterSegment string `json:"clusterSegment,omitempty"`
 }
 
+// RevisionInfo defines the revision information.
 type RevisionInfo struct {
+	// RevisionStatus communicates the observed state of the Revision (from the controller).
+	// +optional
 	RevisionStatus knativev1.RevisionStatus `json:"RevisionsStatus,omitempty"`
-	RevisionName   string                   `json:"name,omitempty"`
+
+	// RevisionName defines the name of the revision.
```

Comment:
Does it _define_ the name of the revision, or is just the name of the current revision?